packages/experiments/src/components/Toggle/Toggle.types.ts

```diff
@@ -88,39 +100,35 @@ export type IToggleViewProps = Pick<IToggleProps, 'as' | 'label' | 'ariaLabel' |
      * Text to display next to the toggle.
      */
     text?: string;
+
+    /**
+     * Reference to the toggle button.
+     */
+    toggleButtonRef?: React.RefObject<HTMLButtonElement>;
   };
 
 /**
  * Styles for the Toggle component.
  */
-export interface IToggleStyles {
-  /**
-   * Root element.
-   */
-  root: IStyle;
-
-  /**
-   * Label element above the toggle.
-   */
-  label: IStyle;
-
-  /**
-   * Container for the toggle pill and the text next to it.
-   */
-  container: IStyle;
-
-  /**
-   * Pill, rendered as a button.
-   */
-  pill: IStyle;
+export interface IToggleStyleVariablesTypes {
+  pillBackground?: string;
+  pillHoveredBackground?: string;
+  pillBorderColor?: string;
+  pillHoveredBorderColor?: string;
+  pillJustifyContent?: IRawStyleBase['justifyContent'];
+  pillHighContrastBackground?: string;
+  pillHighContrastHoveredBackground?: string;
+  pillHighContrastBorderColor?: string;
+  pillHighContrastHoveredBorderColor?: string;
+
+  thumbBackground?: string;
+  thumbHighContrastBackground?: string;
+  thumbHighContrastBorderColor?: string;
+
+  textColor?: string;
+  textHighContrastColor?: string;
+}
 
-  /**
-   * Thumb inside of the pill.
-   */
-  thumb: IStyle;
+export type IToggleStyleVariables = { [PState in IToggleStates]?: IToggleStyleVariablesTypes };
```

Comment:
I think this should be:
```ts
IToggleStyleVariableTypes | (props: IToggleViewProps) => IToggleStyleVariableTypes;
```
Which could be abstracted in a `type` definition.

Reply:
In other words, exposing the state in the type I think is confusing. States are not necessarily mutually exclusive, and therefore, what defines which state has a higher precedence?

Example:

```ts
{
  enabled: {
    background: 'red'
  },
  checked: {
     background: 'green'
  }
}
```

What color is it?

Reply:
I didn't catch this earlier, but `checked` here implies `enabled` and states are mutually exclusive. I think it's unintuitive and something we need to work on